cluster-autoscaler/processors/nodegroups/autoprovisioning_processor.go

```diff
@@ -111,3 +113,38 @@ func addAllMachineTypesForConfig(context *context.AutoscalingContext, systemLabe
 	}
 	return nodeGroups
 }
+
+// AutoprovisioningNodeGroupManager is responsible for creating/deleting autoprovisioned node groups.
+type AutoprovisioningNodeGroupManager struct {
+}
+
+// NewAutoprovisioningNodeGroupManager creates an instance of NodeGroupManagementProcessor.
+func NewAutoprovisioningNodeGroupManager() NodeGroupManager {
+	return &AutoprovisioningNodeGroupManager{}
+}
+
+// CreateNodeGroupIfNotExist creates autoprovisioned node group if it doesn't exist.
+func (p *AutoprovisioningNodeGroupManager) CreateNodeGroupIfNotExist(context *context.AutoscalingContext, nodeGroup cloudprovider.NodeGroup) (cloudprovider.NodeGroup, errors.AutoscalerError) {
```

Comment:
I don't like having this in the same file as AutoprovisioningNodeGroupListProcessor. Those are 2 different things that happen to have similar names. Let's put it in separate file. I'd probably prefer to put it in separate folder too.

Reply:
done

Reply:
Or, as @aleksandra-malinowska pointed out to me, we could have a single struct implementing both interfaces. I'm ok with both approaches.

Reply:
Moved to separate file.